contracts/account/utils/draft-ERC4337Utils.sol

```diff
@@ -98,6 +98,16 @@ library ERC4337Utils {
         return result;
     }
 
+    /// @dev Returns `factory` from the {PackedUserOperation}, or address(0) if the initCode is empty or not properly formatted.
+    function factory(PackedUserOperation calldata self) internal pure returns (address) {
+        return self.initCode.length < 20 ? address(0) : address(bytes20(self.initCode[0:20]));
+    }
+
+    /// @dev Returns `factoryData` from the {PackedUserOperation}, or empty bytes if the initCode is empty or not properly formatted.
+    function factoryData(PackedUserOperation calldata self) internal pure returns (bytes calldata) {
+        return self.initCode.length < 20 ? _emptyCalldataBytes() : self.initCode[20:];
```

Comment:
i think doing this is equivalent to the `_emptyCalldataBytes` function
```suggestion
        return self.initCode.length < 20 ? self.initCode[0:0] : self.initCode[20:];
```

Reply:
not sure about gas but i dont expect a huge difference

Reply:
self.initCode is not at the begining of the calldata, so likelly doing that involves:
- copying `self` (that is a call data pointer) 
- pushing some offset (computing the offset may includes some calldataload?)
- adding the two, so that your calldata pointer starts at self.initCode, and has length 0

I'm expecting that to be more expensive. 

I also personally prefer having a very explicit "this is an empty bytes object" function.